packages/runtime/test/integration/pool-transaction-fee-test.ts

```diff
@@ -86,4 +98,85 @@ describe("Pooled Transaction Fees Test", function () {
 			RUNTIME_ERRORS.TRANSACTION.FEES_NOT_ENOUGH
 		);
 	});
+
+	it("Should unfunded accounts be able to issue transactions of group size 3", () => {
+		const amount = 4000;
+		const fee = 3000;
+		// group with fee distribution. Pooled transaction fee
+		const groupTx: types.ExecParams[] = [
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: john.account,
+				toAccountAddr: alice.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: fee }, // this covers fee of entire group txns.
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				toAccountAddr: bob.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee.
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: bob.account,
+				toAccountAddr: john.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee.
+			},
+		];
+
+		runtime.executeTx(groupTx);
+
+		syncAccounts();
+		assert.equal(john.balance(), BigInt(initialBalance) - BigInt(fee));
+		assert.equal(alice.balance(), BigInt(initialBalance));
+		assert.equal(bob.balance(), BigInt(initialBalance));
+	});
+
+	it("Should unfunded accounts be able to issue transactions and opt-in", () => {
+		setupAsset();
+		const amount = 200000;
+		const fee = 3000;
+		// group with fee distribution
+		const groupTx: types.ExecParams[] = [
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: john.account,
+				toAccountAddr: alice.address,
+				amountMicroAlgos: amount,
+				payFlags: { totalFee: 0 }, // with 0 txn fee
+			},
+			{
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				toAccountAddr: elon.address,
```

Comment:
lets not send anything to this account

Reply:
Why so?? This is an unfunded account. This increases test coverage as one of account in group txn is unfunded one.

Reply:
and assert if the balance ==0

Reply:
It's already been checked
`assert.equal(elon.balance(), BigInt(amount)); // unfunded account`

Reply:
Added one more assert to verify.

Reply:
this is to make sure that if the accounts balance is zero it can still issue transaction. If you send something to it it might interpret the account as not unfunded anymore. That's why I suggest no to send anything to this account

Reply:
Yes, assert added.

Reply:
I mean lets not send anything to it